idaes/models_extra/co2_capture_and_utilization/unit_models/membrane_1d.py

```diff
@@ -0,0 +1,321 @@
+            streams=streams_dict,
+            number_of_finite_elements=self.config.finite_elements,
+        )
+
+        self.feed_side_inlet = Port(extends=self.mscontactor.feed_side_inlet)
+        self.feed_side_outlet = Port(extends=self.mscontactor.feed_side_outlet)
+        if self.config.sweep_flow is True:
+            self.sweep_side_inlet = Port(extends=self.mscontactor.sweep_side_inlet)
+        self.sweep_side_outlet = Port(extends=self.mscontactor.sweep_side_outlet)
+
+        self._make_geometry()
+        self._make_performance()
+
+    def _make_geometry(self):
+
+        self.area = Var(
+            initialize=100, units=units.cm**2, doc="Area per cell (or finite element)"
+        )
+
+        self.length = Var(initialize=100, units=units.cm, doc="The membrane length")
+        self.cell_length = Expression(expr=self.length / self.config.finite_elements)
+
+        self.cell_area = Var(initialize=100, units=units.cm**2, doc="The membrane area")
+
+        @self.Constraint()
+        def area_per_cell(self):
+            return self.cell_area == self.area / self.config.finite_elements
+
+    def _make_performance(self):
+        feed_side_units = (
+            self.config.feed_side.property_package.get_metadata().derived_units
+        )
+
+        self.permeance = Var(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            self.mscontactor.feed_side.component_list,
+            initialize=1,
+            doc="Values in Gas Permeance Unit(GPU)",
+            units=units.dimensionless,
+        )
+
+        self.gpu_factor = Param(
+            default=10e-8 / 13333.2239,
+            units=units.m / units.s / units.Pa,
+            mutable=True,
+            doc=" This is a coefficient that will convert the unit of permeability from GPU to SI units for further calculation",
+        )
+
+        """
+        Selectivity is defined for cases where some permeabilities are unavailable. Only define the selectivity
+         between different components; others should be set to 1. If the permeabilities of all components are defined
+         there is no need to define selectivity, as this may overdefine the problem.
+        """
+        self.selectivity = Var(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            self.mscontactor.feed_side.component_list,
+            self.mscontactor.feed_side.component_list,
+            initialize=1,
+            units=units.dimensionless,
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            self.mscontactor.feed_side.component_list,
+            self.mscontactor.feed_side.component_list,
+            doc="permeance calculation",
+        )
+        def permeance_calculation(self, t, e, a, b):
+            return (
+                self.permeance[t, e, a] * self.selectivity[t, e, a, b]
+                == self.permeance[t, e, b]
+            )
```

Comment:
This looks like it results in an over-specified problem. You write a constraint for `a, b` and `b, a` which are reflections of each other. If the user specifies all the selectivities they will have a degree of freedom issue.

I think you really need to only cover the unique pairs (and exclude a=b).